x/ccv/consumer/keeper/relay.go

```diff
@@ -25,7 +24,12 @@ import (
 //
 // Note: CCV uses an ordered IBC channel, meaning VSC packet changes will be accumulated (and later
 // processed by ApplyCCValidatorChanges) s.t. more recent val power changes overwrite older ones.
-func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, newChanges ccv.ValidatorSetChangePacketData) exported.Acknowledgement {
+func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, newChanges ccv.ValidatorSetChangePacketData) error {
+	// validate packet data upon receiving
+	if err := newChanges.ValidateBasic(); err != nil {
```

Comment:
Consider calling this `Validate` and not `ValidateBasic`

By convention, `ValidateBasic` is used on Tx Message types. This is a type of packet so the naming can be confusing.

Reply:
Good call. Especially as this function is not (yet) called on `checkTx`, while `ValidateBasic` usually is. See https://github.com/cosmos/interchain-security/pull/1460/commits/5488484374096426ed33c0cb76aacba3313f15b6.